src/main/scala/com/databricks/labs/overwatch/utils/UpgradeV2.scala

```diff
@@ -0,0 +1,405 @@
+import com.databricks.labs.overwatch.env.Workspace
+import com.databricks.labs.overwatch.pipeline.TransformFunctions._
+import com.databricks.labs.overwatch.pipeline._
+import com.databricks.labs.overwatch.utils.Helpers.{fastDrop, parOptimize}
+import com.databricks.labs.overwatch.utils.Upgrade.spark
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions.col
+import spark.implicits._
+
+import java.util.concurrent.ConcurrentHashMap
+import scala.collection.JavaConverters.mapAsScalaConcurrentMapConverter
+import scala.collection.concurrent
+import scala.collection.mutable.ArrayBuffer
+
+class UpgradeTo0610(prodWorkspace: Workspace, startStep: Int = 1, endStep: Int = 3, enableUpgradeBelowDBR104: Boolean = false) extends UpgradeHandler {
+
+
+  def this(workspace: Workspace) {
+    this(workspace, 1, 3,false)
+  }
+
+
+  override def upgrade(): DataFrame = {
+    val tempDir = s"/tmp/overwatch/upgrade0610_status__ctrl_0x111/${System.currentTimeMillis()}"
+    val etlDatabaseName = prodWorkspace.database.getDatabaseName
+    dbutils.fs.mkdirs(tempDir) // init tempDir -- if no errors it wouldn't be created
+    val blankConfig = new Config()
+    val currentSchemaVersion = SchemaTools.getSchemaVersion(etlDatabaseName)
+    val numericalSchemaVersion = getNumericalSchemaVersion(currentSchemaVersion)
+    val targetSchemaVersion = "0.610"
+    validateSchemaUpgradeEligibility(currentSchemaVersion, targetSchemaVersion)
+    validateNumericalSchemaVersion(numericalSchemaVersion, 600, 610)
+    val upgradeStatus: ArrayBuffer[UpgradeReport] = ArrayBuffer()
+    val dbrVersion = spark.conf.get("spark.databricks.clusterUsageTags.effectiveSparkVersion")
+    val dbrMajorV = dbrVersion.split("\\.").head
+    val dbrMinorV = dbrVersion.split("\\.")(1)
+    val dbrVersionNumerical = s"$dbrMajorV.$dbrMinorV".toDouble
+    val initialSourceVersions: concurrent.Map[String, Long] = new ConcurrentHashMap[String, Long]().asScala
+    val packageVersion = blankConfig.getClass.getPackage.getImplementationVersion.replaceAll("\\.", "").tail.toInt
+    val startingSchemaVersion = SchemaTools.getSchemaVersion(etlDatabaseName).split("\\.").takeRight(1).head.toInt
+    validateSchemaAndPackageVersion(startingSchemaVersion, packageVersion, 600, 610)
+
+    (startStep to endStep).foreach {
+      case 1 =>
+        val stepMsg = Some("Step 1: Upgrade Schema - Job Status Silver")
+        println(stepMsg.get)
+        logger.log(Level.INFO, stepMsg.get)
+        val targetName = "job_status_silver"
+        try {
+          checkIfTargetExists(etlDatabaseName, targetName)
+          initialSourceVersions.put(targetName, Helpers.getLatestTableVersionByName(s"${etlDatabaseName}.${targetName}"))
+          val jobSilverDF = spark.table(s"${etlDatabaseName}.${targetName}")
+          removeNestedColumnsAndSaveAsTable(jobSilverDF,"new_settings", Array("tasks", "job_clusters"),etlDatabaseName,targetName)
```

Comment:
This should be the entirety of the step 1 upgrade func for 0610 upgrade. This hopefully will be all the code necessary to complete the 0610 step 1 upgrade as a dev building the upgrade.